lib/AST/ASTImporter.cpp

```diff
@@ -5660,16 +5660,37 @@ ASTNodeImporter::VisitFunctionTemplateDecl(FunctionTemplateDecl *D) {
         // name conflict), because FunctionTemplates can possibly overload.
       } // template
     }   // for
-  }
 
-  auto ParamsOrErr = import(D->getTemplateParameters());
-  if (!ParamsOrErr)
-    return ParamsOrErr.takeError();
+#if 0
+    // alternative solution
+    if (FoundByLookup) {
+      if (dyn_cast<CXXMethodDecl>(FoundByLookup->getTemplatedDecl())) {
+        if (D->getLexicalDeclContext() == D->getDeclContext()) {
+          if (!D->getTemplatedDecl()->doesThisDeclarationHaveABody()) {
+            return Importer.MapImported(D, const_cast<FunctionTemplateDecl *>(FoundByLookup));
+          } else {
+            // Let's continue and build up the redecl chain in this case.
+            // FIXME Merge the functions into one decl.
+          }
+        }
+      }
+    }
+#endif
+  }
```

Comment:
We could simplify the code here because this condition will always be true: ` (!LexicalDC->isFunctionOrMethod())`. Actually, we should assert on this condition.
This is because it is illegal to declare/define any function templates in block scope.